### PR TITLE
Delete nil esdt data

### PR DESCRIPTION
**Pull request by sasurobert**
Delete nil esdt data

### CODE REVIEW

File: process/smartContract/builtInFunctions/esdtTransfer.go [hunk 1]
```diff
@@ -265,12 +265,26 @@ func checkFrozeAndPause(
...
+
 func saveESDTData(
 	userAcnt state.UserAccountHandler,
 	esdtData *esdt.ESDigitalToken,
 	key []byte,
 	marshalizer marshal.Marshalizer,
 ) error {
+	isValueZero := esdtData.Value.Cmp(zero) == 0
+	if isValueZero && arePropertiesEmpty(esdtData.Properties) {
+		return userAcnt.DataTrieTracker().SaveKeyValue(key, nil)
```
Comment (bogdan-rosianu) [inline process/smartContract/builtInFunctions/esdtTransfer.go:285]: will this remove the key also? or the key remains but with `null` value?

File: process/smartContract/builtInFunctions/esdtTransfer.go [hunk 2]
```diff
@@ -265,12 +265,26 @@ func checkFrozeAndPause(
 	return nil
 }
 
+func arePropertiesEmpty(properties []byte) bool {
+	for _, property := range properties {
```
Comment (bogdan-rosianu) [inline process/smartContract/builtInFunctions/esdtTransfer.go:269]: could use an early `if len(properties) > 0 { return false }`

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

File: process/smartContract/builtInFunctions/esdtTransfer.go (hunk 1 above)
Comment (sasurobert) [reply process/smartContract/builtInFunctions/esdtTransfer.go:285]: this removes the key.

File: process/smartContract/builtInFunctions/esdtTransfer.go (hunk 2 above)
Comment (sasurobert) [reply process/smartContract/builtInFunctions/esdtTransfer.go:269]: no. as if you do len(make([]byte, 10)) it returns 10

**Pull request merged**